tests/alpaka/alpaka_main.cpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Alpaka include(s).
+#include <alpaka/alpaka.hpp>
+#include <alpaka/example/ExampleDefaultAcc.hpp>
+
+// GoogleTest include(s).
+#include <gtest/gtest.h>
+
+#include <cstdint>
+#include <iostream>
+
+template <typename Acc>
+ALPAKA_FN_ACC float process(Acc const& acc, uint32_t idx) {
+    return alpaka::math::sin(acc, idx);
+}
+
+struct VectorOpKernel {
+    template <typename Acc>
+    ALPAKA_FN_ACC void operator()(Acc const& acc, float* result,
+                                  uint32_t n) const {
+        using namespace alpaka;
+        auto const globalThreadIdx =
+            getIdx<alpaka::Grid, alpaka::Threads>(acc)[0u];
+
+        // It is possible to get index type from the accelerator,
+        // but for simplicity we just repeat the type here
+        if (globalThreadIdx < n) {
+            result[globalThreadIdx] = process(acc, globalThreadIdx);
+        }
+    }
+};
+
+/// Copy vector to device, set element[i] = sin(i) on device, then compare with
+/// same calculation on host
+GTEST_TEST(AlpakaBasic, VectorOp) {
+    using namespace alpaka;
+    using Dim = DimInt<1>;
+    using Idx = uint32_t;
+
+    using Acc = ExampleDefaultAcc<Dim, Idx>;
+    std::cout << "Using alpaka accelerator: " << alpaka::getAccName<Acc>()
+              << std::endl;
```

Comment:
Please remove this printout. It makes the output of the test "less clean than I'd like". :stuck_out_tongue:

```
[bash][celeborn]:build > ./bin/traccc_test_alpaka 
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from AlpakaBasic
[ RUN      ] AlpakaBasic.VectorOp
Using alpaka accelerator: AccCpuSerial<1,j>
[       OK ] AlpakaBasic.VectorOp (0 ms)
[----------] 1 test from AlpakaBasic (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (0 ms total)
[  PASSED  ] 1 test.
[bash][celeborn]:build >
```